### PR TITLE
Closes: #5395 - Incorrect permission in navigation

### LINKED ISSUE #5395: Incorrect permission assigned to Role in navigation
### Environment
* Nautobot version (Docker tag too if applicable): 2.1.X
* Python version: NA
* Database platform, version: NA
* Middleware(s): NA

### Steps to Reproduce
1. Log in with user that does not have view permissions to role BUT has at least 1 permission for an object in the organization menu
2. expand organization
3. click Role

### Expected Behavior
Role not visible in navigation

### Observed Behavior
Role visible in navigation and is clickable, on click the role list view returns access denied as expected

### PR DESCRIPTION
**Pull request by whitej6**
# Closes #5395
# What's Changed
Updated to use correct permission
# Screenshots

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/5395.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed use of invalid permission for extras.Role in navigation.py
```
Comment (glennmatthews) [inline changes/5395.fixed:1]:
```suggestion
Fixed use of incorrect permission for extras.Role in navigation.py.
```

**Review by glennmatthews — APPROVED** [review]
Thanks!

**Review by HanlinMiao — APPROVED** [review]

**Comment by glennmatthews** [comment]
Now fixes #5380 as well and I added a test case to catch any similar issues in the future.

File: nautobot/core/tests/test_navigations.py [hunk 2]
```diff
@@ -1,22 +1,79 @@
...
+from nautobot.core.choices import ButtonActionColorChoices, ButtonActionIconChoices
+from nautobot.core.templatetags.helpers import bettertitle
+from nautobot.core.utils.lookup import get_route_for_model
+from nautobot.core.utils.permissions import get_permission_for_model
+from nautobot.extras.registry import registry
 
 
-# TODO(timizuo): Here might not be the best place to add this test class
+@tag("unit")
+class NavMenuTestCase(TestCase):
```
Comment (HanlinMiao) [inline nautobot/core/tests/test_navigations.py:15]: Love the test!

**Review by HanlinMiao — APPROVED** [review]

File: changes/5395.fixed [hunk 3]
```diff
@@ -0,0 +1 @@
+Fixed use of incorrect permission for extras.Role in navigation.py.
```
[the remark this answers is not in the document]
Comment (glennmatthews) [reply changes/5395.fixed:1]: Perhaps:

```suggestion
Fixed incorrect permission for "Roles" link in the navigation menu.
```

[the remark this answers is not in the document]
Comment (gsnider2195) [reply changes/5395.fixed:1]: I like it

**Pull request merged**